frontend/src/metabase/lib/react-to-html.tsx

```diff
@@ -0,0 +1,19 @@
+import type { ReactNode } from "react";
+import { flushSync } from "react-dom";
+import { createRoot } from "react-dom/client";
+
+/**
+ * Given a React node, it renders the node into a HTML string synchronously.
+ *
+ * https://18.react.dev/reference/react-dom/server/renderToString#removing-rendertostring-from-the-client-code
+ */
+export function reactNodeToHtmlString(node: ReactNode) {
```

Comment:
Btw, we also have `frontend/src/metabase/lib/react-compat.ts`

Reply:
@npretto technically yes, but we are already incompatible with React 17 because of Mantine 7 so there's not much point 😞